sihl/src/database_migration_repo.ml

```diff
@@ -107,81 +119,48 @@ module MariaDb : Sig = struct
          version = VALUES(version),
          dirty = VALUES(dirty)
        |sql}
+         table)
   ;;
 
-  let upsert ~state =
+  let upsert table state =
     Database.query (fun (module Connection : Caqti_lwt.CONNECTION) ->
-        Connection.exec upsert_request (Migration.to_tuple state)
+        Connection.exec (upsert_request table) (Migration.to_tuple state)
         |> Lwt.map Database.raise_error)
   ;;
 end
 
 module PostgreSql : Sig = struct
   module Migration = Migration
 
-  let create_request =
+  let create_request table =
     Caqti_request.exec
       Caqti_type.unit
-      {sql|
-       CREATE TABLE IF NOT EXISTS core_migration_state (
+      (Format.sprintf
+         {sql|
+       CREATE TABLE IF NOT EXISTS %s (
```

Comment:
Could this be an attack vector for SQL injection?

Reply:
No, the table name is not based on any end user input. You need access to the env variables to set it.